### PR TITLE
Create question endpoint

### PR DESCRIPTION
**Pull request by a5x55555**
namespace fix

### CODE REVIEW

File: src/ExamService.Business/Question/CreateQuestionCommand.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateQuestionRequest request)
+    {
+      if (!_validator.ValidateCustom(request, out List<string> errors) || !request.ExamId.HasValue)
+      {
+        return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.BadRequest, errors);
+      }
+
+      OperationResultResponse<Guid?> response = new();
+
+      response.Body = await _repository.CreateAsync(_mapper.Map(request, request.ExamId.Value));
```
Comment (SolovyovDmitry) [inline src/ExamService.Business/Question/CreateQuestionCommand.cs:48]: what's the point to pass examId in mapper separately if you have it in request itself which you also pass?

File: src/ExamService.Validation/Answer/CreateAnswerRequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace LT.DigitalOffice.ExamService.Validation.Answer
+{
+  public class CreateAnswerRequestValidator : AbstractValidator<CreateAnswerRequest>, ICreateAnswerRequestValidator
+  {
+    public CreateAnswerRequestValidator()
+    {
+      RuleFor(request => request.Answer)
+        .Cascade(CascadeMode.Stop)
+        .MinimumLength(0).WithMessage("Minimum length is 0")
```
Comment (SolovyovDmitry) [inline src/ExamService.Validation/Answer/CreateAnswerRequestValidator.cs:12]: Can it be less? And shouldn't there be not empty/null check?

File: src/ExamService.Business/Question/CreateQuestionCommand.cs (hunk 1 above)
Comment (a5x55555) [reply src/ExamService.Business/Question/CreateQuestionCommand.cs:48]: The reason is not to write an additional method in mapper which is already used in the other request.

File: src/ExamService.Validation/Answer/CreateAnswerRequestValidator.cs (hunk 2 above)
Comment (a5x55555) [reply src/ExamService.Validation/Answer/CreateAnswerRequestValidator.cs:12]: This field goes with "Required" attribute by default, i think it gives me an opportunity to not check if it's null.

File: src/ExamService.Business/Question/CreateQuestionCommand.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+{
+  public class CreateQuestionCommand : ICreateQuestionCommand
+  {
+    private readonly IDbQuestionMapper _mapper;
+    private readonly IQuestionRepository _repository;
+    private readonly ICreateQuestionRequestValidator _validator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IResponseCreator _responseCreator;
+
+    public CreateQuestionCommand(IDbQuestionMapper mapper,
```
Comment (MariyaSin) [inline src/ExamService.Business/Question/CreateQuestionCommand.cs]: move mapper to next line

File: src/ExamService.Validation/Answer/CreateAnswerRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+using LT.DigitalOffice.ExamService.Validation.Answer.Interfaces;
+
+namespace LT.DigitalOffice.ExamService.Validation.Answer
+{
+  public class CreateAnswerRequestValidator : AbstractValidator<CreateAnswerRequest>, ICreateAnswerRequestValidator
+  {
+    public CreateAnswerRequestValidator()
+    {
+      RuleFor(request => request.Answer)
+        .Cascade(CascadeMode.Stop)
```
Comment (MariyaSin) [inline src/ExamService.Validation/Answer/CreateAnswerRequestValidator.cs]: no need use cascade here

File: src/ExamService.Validation/Question/CreateQuestionRequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace LT.DigitalOffice.ExamService.Validation.Question
+{
+  public class CreateQuestionRequestValidator : AbstractValidator<CreateQuestionRequest>, ICreateQuestionRequestValidator
+  {
+    public CreateQuestionRequestValidator(ICreateAnswerRequestValidator answerValidator)
+    {
+      RuleFor(request => request.Subject)
+        .Cascade(CascadeMode.Stop)
+        .NotEmpty().WithMessage("Subject mustn't be empty")
```
Comment (MariyaSin) [inline src/ExamService.Validation/Question/CreateQuestionRequestValidator.cs]: check on 15 line is enougth. Remove 13-14 lines

**Review by SolovyovDmitry — APPROVED** [review]

File: src/ExamService.Business/Question/CreateQuestionCommand.cs (hunk 1 above)
Comment (SolovyovDmitry) [reply src/ExamService.Business/Question/CreateQuestionCommand.cs:48]: Sure it was pointless even before but I guess it's not a big deal to change if @MariyaSin doesn't mind

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**